custom_components/pyscript/trigger.py

```diff
@@ -644,6 +644,112 @@ def start(self):
+            if check_var == var_name and old_value != value:
+                _LOGGER.debug(
+                    "%s ident_any change detected at state",
+                    self.name,
+                )
+                return True
+
+            if check_var.startswith(f"{var_name}."):
+                var_pieces = check_var.split('.')
+                if len(var_pieces) == 3 and f"{var_pieces[0]}.{var_pieces[1]}" == var_name:
+                    if var_pieces[2] == "*":
+                        # catch all has been requested, check all attributes for change
+                        all_attributes = set()
+                        if value is not None:
+                            all_attributes |= set(value.__dict__.keys())
+                        if old_value is not None:
+                            all_attributes |= set(old_value.__dict__.keys())
+                        all_attributes -= {"last_updated", "last_changed"}
+                        for attribute in all_attributes:
+                            attrib_val = getattr(value, attribute, None)
+                            attrib_old_val = getattr(old_value, attribute, None)
+                            if attrib_old_val != attrib_val:
+                                _LOGGER.debug(
+                                    "%s ident_any change detected in * at %s",
+                                    self.name,
+                                    attribute,
+                                )
+                                return True
+                    else:
+                        attrib_val = getattr(value, var_pieces[2], None)
+                        attrib_old_val = getattr(old_value, var_pieces[2], None)
+                        if attrib_old_val != attrib_val:
+                            _LOGGER.debug(
+                                "%s ident_any change detected at %s",
+                                self.name,
+                                var_pieces[2],
+                            )
+                            return True
+
+        _LOGGER.debug(
+            "%s no ident_any change detected",
+            self.name,
+        )
+        return False
+
+    def ident_values_changed(self, func_args):
+        """Check for changes to state or attributes on ident vars"""
+        value = func_args.get('value')
+        old_value = func_args.get('old_value')
+        var_name = func_args.get('var_name')
+
+        if var_name is None:
+            _LOGGER.debug(
+                "%s ident changes not detected because no var_name",
+                self.name,
+            )
+            return False
+
+        for check_var in self.state_trig_ident:
+            if check_var in self.state_trig_ident_any:
+                _LOGGER.debug(
+                    "%s ident change skipping %s because also ident_any",
+                    self.name,
+                    check_var,
+                )
+                continue
```

Comment:
Is this whole `if` clause necessary?  In this case `ident_any_values_changed()` would already have returned `True`.

Reply:
My thought process was that ident_any_values_changed() checked it, but the value did not change. Therefore, it didn't return True. So, in ident_values_changed() we can skip it (and save some CPU) since they were already checked. If we take the if out, since it didn't return True before, it won't return True now either.